src/django_components/template_loader.py

```diff
@@ -26,3 +26,10 @@ def get_dirs(self, include_apps: bool = True) -> List[Path]:
         `BASE_DIR` setting is required.
         """
         return get_component_dirs(include_apps)
+
+
+# NOTE: Django's template loaders have the pattern of using the `Loader` class name.
+#       However, this then makes it harder to track and distinguish between different loaders.
+#       So internally we use the name `DjcLoader` instead.
+#       But for public API we use the name `Loader` to match Django.
+Loader = DjcLoader
```

Comment:
My initial implementation relied on passing extra data through our template loader, which forced me to debug how template loaders work and whatnot.

As part of that I renamed it from `Loader` to `DjcLoader`, so it's easier to distinguish from other Loaders when debugging.

To remain backwards compatible, and inline with how Django's template loaders are defined, this file still defines a `Loader` alias.